app/src/main/java/com/duckduckgo/app/browser/certificates/rootstore/TrustedCertificateStore.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2020 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.certificates.rootstore
+
+import android.net.http.SslCertificate
+import com.duckduckgo.app.browser.certificates.toX509Certificate
+import com.duckduckgo.app.browser.certificates.CertificateType
+import com.duckduckgo.app.browser.certificates.LetsEncryptCertificateProvider
+import timber.log.Timber
+import java.security.cert.*
+
+interface TrustedCertificateStore {
+    fun validateSslCertificateChain(sslCertificate: SslCertificate)
+}
+
+class TrustedCertificateStoreImpl(
+    private val letsEncryptCertificateProvider: LetsEncryptCertificateProvider
+) : TrustedCertificateStore {
+
+    /**
+     * Validates the given SSL certificate chaing
+     * @param sslCertificate the SSL certificate to be validated
+     *
+     * @throws CertificateExpiredException when any issuer in the [sslCertificate] chain is expired
+     * @throws CertificateNotYetValidException when any issuer in the [sslCertificate] chain is not yet valid
+     * @throws CertificateException when we could not validate the [sslCertificate] certificate chain
+     */
+    override fun validateSslCertificateChain(sslCertificate: SslCertificate) {
```

Comment:
Yeah I agree, did that because `checkValidity` throws so I thought I would not break that pattern. But I was not very happy about it either so I'll change it